public/app/features/serviceaccounts/CreateTokenModal.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState } from 'react';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import {
+  Button,
+  ClipboardButton,
+  DatePickerWithInput,
+  Field,
+  FieldSet,
+  HorizontalGroup,
+  Icon,
+  Input,
+  Label,
+  Modal,
+  RadioButtonGroup,
+  useStyles2,
+} from '@grafana/ui';
+
+const expirationOptions = [
+  { label: 'No expiration', value: false },
+  { label: 'Set expiration date', value: true },
+];
+
+interface CreateTokenModalProps {
+  isOpen: boolean;
+  token: string;
+  onCreateToken: (name: string) => void;
+  onClose: () => void;
+}
+
+export const CreateTokenModal = ({ isOpen, token, onCreateToken, onClose }: CreateTokenModalProps) => {
+  const [newTokenName, setNewTokenName] = useState('');
+  const [isWithExpirationDate, setIsWithExpirationDate] = useState(false);
+  const [newTokenExpirationDate, setNewTokenExpirationDate] = useState<Date | string>('');
+  const [isExpirationDateValid, setIsExpirationDateValid] = useState(false);
+  const styles = useStyles2(getStyles);
+
+  const onExpirationDateChange = (value: Date | string) => {
+    const isValid = value !== '';
+    setIsExpirationDateValid(isValid);
+    setNewTokenExpirationDate(value);
+  };
+
+  const onCloseInternal = () => {
+    setNewTokenName('');
+    setIsWithExpirationDate(false);
+    setNewTokenExpirationDate('');
+    setIsExpirationDateValid(false);
+    onClose();
+  };
+
+  const modalTitle = (
+    <div className={styles.modalHeaderTitle}>
+      <Icon className={styles.modalHeaderIcon} name="key-skeleton-alt" size="lg" />
+      <span>{!token ? 'Add service account token' : 'Service account token created'}</span>
+    </div>
+  );
+
+  return (
+    <Modal isOpen={isOpen} title={modalTitle} onDismiss={onCloseInternal} className={styles.modal}>
+      {!token ? (
+        <>
+          <FieldSet>
+            <Field
+              label="Display name"
+              description="Optional name to easily identify the token"
+              className={styles.modalRow}
+            >
+              <Input
+                name="tokenName"
+                value={newTokenName}
+                onChange={(e) => {
+                  setNewTokenName(e.currentTarget.value);
+                }}
+              />
+            </Field>
+            <RadioButtonGroup
+              className={styles.modalRow}
+              options={expirationOptions}
+              value={isWithExpirationDate}
+              onChange={setIsWithExpirationDate}
+              size="md"
+            />
+            {isWithExpirationDate && (
+              <Field label="Expiration date" className={styles.modalRow}>
+                <DatePickerWithInput onChange={onExpirationDateChange} value={newTokenExpirationDate} placeholder="" />
```

Comment:
Is the empty placeholder prop necessary here?
```suggestion
                <DatePickerWithInput onChange={onExpirationDateChange} value={newTokenExpirationDate}  />
```